pkg/scalers/aws_common.go

```diff
@@ -29,35 +31,35 @@ type awsConfigMetadata struct {
 	awsAuthorization awsAuthorizationMetadata
 }
 
-func getAwsConfig(awsRegion string, awsEndpoint string, awsAuthorization awsAuthorizationMetadata) (*session.Session, *aws.Config) {
+func getAwsConfig(awsRegion string, awsEndpoint string, awsAuthorization awsAuthorizationMetadata) (*aws.Config, error) {
+	ctx := context.TODO()
 	metadata := &awsConfigMetadata{
 		awsRegion:        awsRegion,
 		awsEndpoint:      awsEndpoint,
-		awsAuthorization: awsAuthorization}
-
-	sess := session.Must(session.NewSession(&aws.Config{
-		Region:   aws.String(metadata.awsRegion),
-		Endpoint: aws.String(metadata.awsEndpoint),
-	}))
+		awsAuthorization: awsAuthorization,
+	}
 
+	configOptions := make([]func(*config.LoadOptions) error, 0)
+	configOptions = append(configOptions, config.WithRegion(metadata.awsRegion))
```

Comment:
Have we lost the `awsEndpoint` usage?

Reply:
I will check this and see how this has to be supplied. Also, we need to make this as optional field.

Reply:
Added changes to use this endpoint.